go-controller/pkg/node/obj_retry_node.go

```diff
@@ -187,6 +210,31 @@ func (h *nodeEventHandler) UpdateResource(oldObj, newObj interface{}, _ bool) er
 		return h.nc.reconcileConntrackUponEndpointSliceEvents(
 			oldEndpointSlice, newEndpointSlice)
 
+	case factory.NodeType:
+		oldNode := oldObj.(*corev1.Node)
+		newNode := newObj.(*corev1.Node)
+
+		// if it's our node that is changing, then resync routes/nftables for all nodes
+		if newNode.Name == h.nc.name {
+			nodes, err := h.nc.watchFactory.GetNodes()
+			if err != nil {
+				return fmt.Errorf("error retrieving nodes: %w", err)
+			}
+			for _, node := range nodes {
+				if node.Name == h.nc.name {
+					continue
+				}
+				if err := h.nc.addOrUpdateNode(node); err != nil {
+					return fmt.Errorf("error updating node %s: %w", node.Name, err)
+				}
+			}
+			return nil
+		}
+
+		// remote node that is changing
+		h.nc.deleteNode(oldNode)
```

Comment:
why delete first? couldnt that cause us to leak out messages?

Reply:
no real reason from a functional perspective. I guess from a code perspective delete does not throw error, so we can at least delete and clean up if addOrUpdate errors out.

Reply:
done